exporter/elasticsearchexporter/integrationtest/datareceiver.go

```diff
@@ -40,22 +54,30 @@ func newElasticsearchDataReceiver(t testing.TB) *esDataReceiver {
 	}
 }
 
-func (es *esDataReceiver) Start(_ consumer.Traces, _ consumer.Metrics, lc consumer.Logs) error {
+func (es *esDataReceiver) Start(tc consumer.Traces, _ consumer.Metrics, lc consumer.Logs) error {
 	factory := receiver.NewFactory(
 		component.MustNewType("mockelasticsearch"),
 		createDefaultConfig,
 		receiver.WithLogs(createLogsReceiver, component.StabilityLevelDevelopment),
+		receiver.WithTraces(createTracesReceiver, component.StabilityLevelDevelopment),
 	)
+	esURL, err := url.Parse(es.endpoint)
+	if err != nil {
+		return fmt.Errorf("invalid ES URL specified %s: %w", es.endpoint, err)
+	}
 	cfg := factory.CreateDefaultConfig().(*config)
-	cfg.ESEndpoint = es.endpoint
+	cfg.ServerConfig.Endpoint = esURL.Host
 
-	var err error
 	set := receivertest.NewNopCreateSettings()
 	// Use an actual logger to log errors.
 	set.Logger = zap.Must(zap.NewDevelopment())
 	es.receiver, err = factory.CreateLogsReceiver(context.Background(), set, cfg, lc)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to create logs receiver: %w", err)
+	}
+	es.receiver, err = factory.CreateTracesReceiver(context.Background(), set, cfg, tc)
```

Comment:
This is overwriting `es.receiver` which is a `receiver.Logs`. Is this intended?

Reply:
Yeah, both will be same since we use `SharedComponent`

Reply:
It might have been a source of confusion for others so I added a comment and assertion (overkill maybe? - feedbacks welcomed) to check if they point to the same object.